lib/raygun.messageBuilder.ts

```diff
@@ -31,23 +31,26 @@ type UserMessageData = RawUserData | string | undefined;
 const humanString = require("object-to-human-string");
 const packageDetails = require("../package.json");
 
-function filterKeys(obj: { [key: string]: any }, filters: string[]) {
+function filterKeys(obj: object, filters: string[]): object {
   if (!obj || !filters || typeof obj !== "object") {
     return obj;
   }
+  // Make temporary copy of the object to avoid mutating the original
+  // Cast to Record<string, object> to enforce type check and avoid using any
+  const _obj = { ...obj } as Record<string, object>;
   Object.keys(obj).forEach(function (i) {
     if (filters.indexOf(i) > -1) {
-      delete obj[i];
+      delete _obj[i];
     } else {
-      obj[i] = filterKeys(obj[i], filters);
+      _obj[i] = filterKeys(_obj[i], filters);
```

Comment:
What will happen if there a cycle in the object? Will we get a Stack Overflow if an object down the line, points back to a parent object? Can we add a test for this, if none exists?

Reply:
That is a good point. I have seen a similar (in the grand scheme of things) problem in the past with another provider where error objects and structures have all of a sudden introduced unexpected self-referencing or issues when serialising.

:+1:  for test

Reply:
Is this a blocker for the release?

Reply:
`filterKeys` is only being used to filter the debug information from the http request, as seen here:

```
queryString: filterKeys(request.query, this._filters),
headers: filterKeys(request.headers, this._filters),
form: filterKeys(request.body, this._filters),
```

It is not being used on error structures, but only on the original request query, headers and body, so I don't expect a recursive object there.

Please let me know if this needs to be researched now or should I create a ticket to look it up in the future.

Reply:
Solution: we all agreed to not deal with this potential (albeit unlikely) issue and will look into it post-release.